### PR TITLE
#61 Added pagination

### LINKED ISSUE #61: Pagination
Apply pagination in all GET methods where is returned list of items.

### PR DESCRIPTION
**Pull request by Prestionyk**
Should close #61

### CODE REVIEW

File: Core/Common/PaginationHelper.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+{
+    public class PaginationHelper
+    {
+
+        public static PagedResponse<T> CreatePagedReponse<T>(T pagedData, PaginationFilter validFilter, int totalRecords, IUriService uriService, string route)
+        {
+
+            var respose = new PagedResponse<T>(pagedData, validFilter.PageNumber, validFilter.PageSize);
+            var totalPages = validFilter.PageSize == 0 ? 1 : ((double)totalRecords / (double)validFilter.PageSize);
+            int roundedTotalPages = Convert.ToInt32(Math.Ceiling(totalPages));
```
Comment (wikhag) [inline Core/Common/PaginationHelper.cs]: You can shorten it to 
`var totalPages = validFilter.PageSize == 0 ? 1 : (int)Math.Ceiling((double)totalRecords / validFilter.PageSize);`

File: Core/Repositories/PaginationRepository.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                    .ToListAsync(),
+                    Quantity = totalRecords
+                };
+            }
+
+            return new ExtendedData<IEnumerable<T>>()
+            {
+                Data = await _context.Set<T>().ToListAsync(),
+                Quantity = totalRecords
+            };
```
Comment (wikhag) [inline Core/Repositories/PaginationRepository.cs]: You can change it to use 1 return and ignore redundancy

**Review by wikhag — CHANGES_REQUESTED** [review]

File: Core/Repositories/PaginationRepository.cs (hunk 2 above)
Comment (Prestionyk) [reply Core/Repositories/PaginationRepository.cs]: That's true

File: Core/Common/PaginationHelper.cs (hunk 1 above)
Comment (Prestionyk) [reply Core/Common/PaginationHelper.cs]: Okey, will be done

**Review by wikhag — APPROVED** [review]

**Review by B34TER — APPROVED** [review]

**Review by Justyn-98 — APPROVED** [review]

**Pull request merged**